### DISCUSSION TITLE
Fdroid version vs play store version

### DISCUSSION THREAD

**Discussion by JulTux** [organicmaps/organicmaps#936] [Q&A]
Feedback after one month of using organicmaps. First of all, congratulations for the work done, it is the only navigation application that is a real open-source alternative to proprietary applications. However, I had to abandon the fdroid version: the localisation was too unstable and inaccurate. This made organicmaps almost unusable for car navigation. The play store version works fine even without the Gapps (I use an AOSP ROM with microg and the Mozilla backend;). Secondly, I would like to make a small remark. I find that organicmaps lacks two things to become a reference: at junctions, it should display the names of directions, road numbers, etc... and traffic information (like MagicEarth for example, which I use daily but which is not open-source). This makes it less practical to use in big cities.
![Screenshot_Mull_20210726-195959_1](https://user-images.githubusercontent.com/68580970/127044640-ee307693-df2e-4ddb-a9e5-8e8563469c36.png)

Are these features being considered?

**Comment by biodranik** [comment]
Can you please clarify in more detail, what was not working in F-Droid version? GPlay version should be almost the same.

Navigation improvements are planned, as they are really useful.

**Reply by JulTux** [reply]
> Navigation improvements are planned, as they are really useful.

Good news !!



> Can you please clarify in more detail, what was not working in F-Droid version?

I will try to clarify...
While navigating in the car, the GPS signal is lost every 4-5 minutes and takes a long time to come back. Often I am located off the road, in the middle of a field or on the next road... The arrow moves forward in a jerky way. Indoors, the location is very approximate.
All this does not happen with the play store version (google service activated with microg. I haven't tried without).

Sorry for my english, it's deepl trad...

**Reply by biodranik** [reply]
So the problem is with GPS service (location), not with localization :) It looks like GPS signal handling should be improved. What about other similar apps? Do they work flawlessly without microG? Google Services provide a faster way to locate you using WiFi and cellular networks, which helps acquire satellites signal faster.

**Reply by JulTux** [reply]
Obviously Magic Earth does not use google services, and it works fine.
Tomorrow I will be driving a lot. I will test both versions on the same road. If this is useful...

**Reply by biodranik** [reply]
Yes, please.

@rtsisyk @vng we have something wrong with our current implementation.

**Reply by JulTux** [reply]
Can I install both versions at the same time ?

**Reply by biodranik** [reply]
No ATM, they have the same ID.

**Comment by JulTux** [comment]
I recorded my screen to show you. I don't know if this is useful for you. On the video, I drive all the time, you can see that the gps signal is unstable. It comes back when I get out of the car on my arrival.
And this happens quite often (every 5-10 min or so, randomly) with the fdroid version, obviously even more often when I'm driving around town; which makes it difficult to use. However, yesterday I drove for 3 hours in the countryside, and there were no bugs.
Today, I drove in town using the play store version (Google services activated) and no bug... Coincidence?

https://user-images.githubusercontent.com/68580970/127397245-077dfcfd-bf86-4f62-92fe-c39d161bb2bf.mp4

**Reply by rtsisyk** [reply]
F-Droid version doesn't include Google Mobile Service and rely purely on hardware GPS location.

**Reply by biodranik** [reply]
Looks like our GPS location code can be improved. It would be great if you can try to use other apps from FDroid and confirm that their GPS is working normally.
